edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Interfaces/IMetricsFileStorage.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// Stores metrics for later upload.
+    /// </summary>
+    public interface IMetricsFileStorage
+    {
+        /// <summary>
+        /// Adds the string to its storage. The current date will be recorded with it.
+        /// </summary>
+        /// <param name="data">data to store.</param>
+        void AddScrapeResult(string data);
```

Comment:
you can write any data to it; why not just name it as WriteData or WriteMetric?

Reply:
Also should it be Async method?

Reply:
renamed, I left it as void since ```File.WriteAllText``` doesn't have an async variant